test/ruby/test_module.rb

```diff
@@ -2890,6 +2890,58 @@ def foo; end
     }
   end
 
+  def test_prepend_constant_lookup
+    m = Module.new do
+      const_set(:C, :m)
+    end
+    c = Class.new do
+      const_set(:C, :c)
+      prepend m
+    end
+    sc = Class.new(c)
+    assert_equal(:m, sc.const_get(:C))
+    assert_equal(:m, sc::C)
+
+    assert_equal(:m, c.const_get(:C))
+    assert_equal(:m, c::C)
+    assert_equal(:m, c.const_get(:C, false))
```

Comment:
@jeremyevans I took a stab at implementing the skipping approach and noticed that in addition to addressing the bug, this PR changes `klass.const_get(:C, false)` to look in prepended modules of `klass`. I find that surprising mainly because I think it prevents users from reading the constant table of `klass`, as the lookup goes to prepended modules.

I took a more conservative approach in #4585 and addressed the bug without this change.